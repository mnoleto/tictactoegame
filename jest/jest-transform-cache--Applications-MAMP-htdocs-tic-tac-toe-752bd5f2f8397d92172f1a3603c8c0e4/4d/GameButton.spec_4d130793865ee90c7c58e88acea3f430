/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_GameButton = require('./GameButton'); /* istanbul ignore next */var _GameButton2 = _interopRequireDefault(_GameButton);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameButton is responsible for render the game grid
// CONTRACTS
// 1. 

describe('GameButton', function () {
  var props = /* istanbul ignore next */void 0,mountedGameButton = /* istanbul ignore next */void 0;
  var gameButton = function gameButton() {
    if (!mountedGameButton) {
      mountedGameButton = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameButton2.default, props));

    }
    return mountedGameButton;
  };

  beforeEach(function () {
    props = {};
    mountedGameButton = undefined;
  });

  it('always render the GameButton', function () {
    var gameButtonElement = gameButton().find('div');
    expect(gameButtonElement.length).toEqual(1);
  });
});