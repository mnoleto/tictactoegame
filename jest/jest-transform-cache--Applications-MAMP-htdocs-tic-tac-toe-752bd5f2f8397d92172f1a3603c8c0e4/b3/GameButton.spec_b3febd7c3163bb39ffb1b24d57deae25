/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_sinon = require('sinon'); /* istanbul ignore next */var _sinon2 = _interopRequireDefault(_sinon);
var /* istanbul ignore next */_GameButton = require('./GameButton'); /* istanbul ignore next */var _GameButton2 = _interopRequireDefault(_GameButton);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameButton is responsible for render the game grid buttons
describe('<GameButton />', function () {
  var props = /* istanbul ignore next */void 0,mountedGameButton = /* istanbul ignore next */void 0;
  var gameButton = function gameButton() {
    if (!mountedGameButton) {
      mountedGameButton = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameButton2.default, props));

    }
    return mountedGameButton;
  };

  beforeEach(function () {
    props = {
      index: 0,
      label: 'E',
      onClickButton: jest.fn() };

    mountedGameButton = undefined;
  });

  it('render the GameButton', function () {
    var gameButtonWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameButton2.default, props));
    expect(gameButtonWrapper).toMatchSnapshot();
  });

  it('render a button as root element', function () {
    var gameButtonElement = gameButton().find('button');
    var wrappingDiv = gameButtonElement.first();
    expect(wrappingDiv.children()).toEqual(gameButton().children());
  });

  it('should have a class game-button', function () {
    var gameButtonElement = gameButton().find('button');
    expect(gameButtonElement.hasClass('game-button')).toBeTruthy();
  });

  it('should have an empty content if gets a prop label with "E" value', function () {
    var gameButtonElement = gameButton().find('button');
    expect(gameButtonElement.text()).toEqual('');
  });

  it('should have display content if gets a prop label different from "E" value', function () {
    props.label = 'X';
    var gameButtonElement = gameButton().find('button');
    expect(gameButtonElement.text()).toEqual('X');
  });

  it('simulate click', function () {
    props.onClickButton = /* istanbul ignore next */_sinon2.default.spy();
    var gameButtonElement = gameButton().find('button');

    gameButtonElement.simulate('click');
    expect(props.onClickButton.callCount).toEqual(1);
  });
});