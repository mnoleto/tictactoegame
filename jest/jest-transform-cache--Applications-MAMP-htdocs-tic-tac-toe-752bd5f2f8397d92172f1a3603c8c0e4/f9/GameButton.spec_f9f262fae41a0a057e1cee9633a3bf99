/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_GameButton = require('./GameButton'); /* istanbul ignore next */var _GameButton2 = _interopRequireDefault(_GameButton);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// GameButton is responsible for render the game grid
// CONTRACTS
// 1. Should render the GameButton

describe('GameButton', function () {
  var props = /* istanbul ignore next */void 0,mountedGameButton = /* istanbul ignore next */void 0;
  var gameButton = function gameButton() {
    if (!mountedGameButton) {
      mountedGameButton = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameButton2.default, props));

    }
    return mountedGameButton;
  };

  beforeEach(function () {
    props = {};
    mountedGameButton = undefined;
  });

  it('render the GameButton', function () {
    var gameButtonWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_GameButton2.default, /* istanbul ignore next */null));
    expect(gameButtonWrapper).toMatchSnapshot();
  });
});