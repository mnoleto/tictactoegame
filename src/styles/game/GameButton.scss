@import '../_colors.scss';

$border-width: 15px;

.game-button {
  background-color: transparent;
  border: 0px;
  border-bottom: $border-width solid $dark-blue;
  border-right: $border-width solid $dark-blue;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 800%;
  font-weight: 700;
  height: 180px;
  padding: 0px;
  width: 32%;
  &:nth-child(3n+3) {
    border-right: 0px solid $dark-blue;
  }
  &:nth-child(1n+7) {
    border-bottom: 0px solid $dark-blue;
  }
  .x {
    color: #000;
  }
  .o {
    color: #fff;
  }
}