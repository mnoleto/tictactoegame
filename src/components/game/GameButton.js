import React, { Component, PropTypes } from 'react';
import '../../styles/game/GameButton.scss';

class GameButton extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { index, onClickButton } = this.props;
    onClickButton(index);
  }

  render() {
    const { label } = this.props;
    const buttonLabel = (label && label === 'E') ? '' : label;
    return (
      <button type="button" className="game-button" onClick={this.handleClick}>{buttonLabel}</button>
    );
  }
};

GameButton.propTypes = {
  label: PropTypes.string,
  index: PropTypes.number,
  onClickButton: PropTypes.func
}

export default GameButton;