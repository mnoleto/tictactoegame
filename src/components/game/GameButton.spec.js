import React from 'react';
import { mount, shallow } from 'enzyme';
import GameButton from './GameButton';

// GameButton is responsible for render the game grid
// CONTRACTS
// 1. Should render the GameButton

describe('GameButton', () => {
  let props, mountedGameButton;
  const gameButton = () => {
    if(!mountedGameButton) {
      mountedGameButton = mount(
        <GameButton {...props} />
      );
    }
    return mountedGameButton;
  };

  beforeEach(() => {
    props = {};
    mountedGameButton = undefined;
  });

  it('render the GameButton', () => {
    const gameButtonWrapper = shallow(<GameButton />);
    expect(gameButtonWrapper).toMatchSnapshot();
  });

  it('render a button as root element', () => {
    let gameButtonElement = gameButton().find('button');
    let wrappingDiv = gameButtonElement.first();
    expect(wrappingDiv.children()).toEqual(gameButton().children());
  });
});