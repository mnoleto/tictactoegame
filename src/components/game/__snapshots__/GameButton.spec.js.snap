exports[`GameButton render the GameButton 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="game-button"
    onClick={[Function]}
    type="button" />,
  "nodes": Array [
    <button
      className="game-button"
      onClick={[Function]}
      type="button" />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GameButton />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GameButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleClick": [Function],
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="game-button"
          onClick={[Function]}
          type="button" />,
        "_debugID": 2,
        "_renderedOutput": <button
          className="game-button"
          onClick={[Function]}
          type="button" />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GameButton />,
}
`;
